std/encoding.d

```diff
@@ -2233,7 +2101,7 @@ body
     }
     else static if(is(Src==AsciiChar))
     {
-        transcode!(char,Dst)(cast(string)s,r);
+        transcodeFromAscii(s, r);
```

Comment:
No, I did mean `@trusted`. But, it doesn't appear that I've sufficiently accounted for how to deal with bugs in the implementation.

On `@trusted` lambdas, when I was thinking clearly about this, I wrote:

> Sometimes a templated function has a memory safe interface only for some instantiations, and uses language constructs disallowed in `@safe`, and that uncheckable code cannot be factored out into a separate function with a memory safe interface, thwarting attribute inference:

In this case, that uncheckable code *can* be factored out into a separate function with a memory safe interface, which is exactly what you initially did... but its utility was confusing (to me, at least) because sans bugs, `transcode` already has a memory safe interface.

I also wrote:

> This approach breaks with the strict guidelines of `@trusted`. This is a necessary evil to achieve `@safe`-ready, but there is little stopping future maintainers from using `p` in an unsafe way. When considering this approach, please make sure that a `@trusted` component with a memory safe interface is not possible, and please leave a comment keeping future maintainers informed of what assumptions need to hold for the code to remain memory safe.

> Do not use this approach for non-templated functions. Between annotating an uncomfortably large swath of code with `@trusted` and breaking the guidelines of `@trusted`, the former is the lesser of two evils.

I still believe the last part which is why I suggested `@trusted` on `transcode` in its entirety. But using a `@trusted` lambda here does not suffer the downsides because it's not actually doing anything memory-unsafe. So I think we should make a distinction between using a `@trusted` nested function for type inference and using it for working around a bug.

So, sorry for the fuss and I think the lambda is fine. The important part, as you know, is of course the comment referencing the bug.